### DISCUSSION TITLE
Reading pointcloud data from a sensor_msgs/PointCloud2 message.

### DISCUSSION THREAD

**Discussion by konsgken** [foxglove/mcap#1100] [General] [answered]
Dear,

I was able to write in a mcap file a sequence of pointcloud data in `sensor_msgs/PointCloud2` messages using the [RosmsgWriter ](https://github.com/foxglove/nuscenes2mcap/blob/9c1826c0e83b9bc5364680e1f7afdd6a7edc199f/RosmsgWriter.py#L7) with [numpy_pc2.array_to_pointcloud2](https://github.com/dimatura/pypcd/blob/20b032bfc729dec853ac810bceeb360f78bdc1d6/pypcd/numpy_pc2.py#L207) from pypcd library. When I try to read the `sensor_msgs/PointCloud2` messages from the mcap file using the [example you provided](https://github.com/foxglove/mcap/blob/fd3ecada218f41929c6d430acba3561e5ebfaf08/python/examples/raw/reader.py) the pointcloud data are serialized (which is what is expected), but I would like to ask you if there is a way to deserialize them and transform them back to numpy  arrays. Is there such a function that I use it?

**Comment by jtbandes** [comment, marked as answer]
Try using the `mcap_ros1` package as described here: https://mcap.dev/guides/python/ros1.html

Or for ROS 2: https://mcap.dev/guides/python/ros2.html

**Reply by konsgken** [reply]
Thanks a lot!
